yarn-project/circuit-types/src/body.ts

```diff
@@ -0,0 +1,179 @@
+    // It seems the da/ tx hash would be fine, would only need to edit circuits ?
+    const newL1ToL2Messages = reader.readVector(Fr);
+
+    // Because TX's in a block are padded to nearest power of 2, this is finding the nearest nonzero tx filled with 1 nullifier
+    const numberOfNonEmptyTxs = calculateNumTxsFromNullifiers(newNullifiers);
+
+    const newEncryptedLogs = reader.readObject(L2BlockL2Logs);
+    const newUnencryptedLogs = reader.readObject(L2BlockL2Logs);
+
+    if (
+      new L2BlockL2Logs(newEncryptedLogs.txLogs.slice(numberOfNonEmptyTxs)).getTotalLogCount() !== 0 ||
+      new L2BlockL2Logs(newUnencryptedLogs.txLogs.slice(numberOfNonEmptyTxs)).getTotalLogCount() !== 0
+    ) {
+      throw new Error('Logs exist in the padded area');
+    }
+
+    const txEffects: TxEffect[] = [];
+
+    const numberOfTxsIncludingEmpty = newNullifiers.length / MAX_NEW_NULLIFIERS_PER_TX;
+
+    for (let i = 0; i < numberOfTxsIncludingEmpty; i += 1) {
+      txEffects.push(
+        new TxEffect(
+          newNoteHashes.slice(i * MAX_NEW_COMMITMENTS_PER_TX, (i + 1) * MAX_NEW_COMMITMENTS_PER_TX),
+          newNullifiers.slice(i * MAX_NEW_NULLIFIERS_PER_TX, (i + 1) * MAX_NEW_NULLIFIERS_PER_TX),
+          newL2ToL1Msgs.slice(i * MAX_NEW_L2_TO_L1_MSGS_PER_TX, (i + 1) * MAX_NEW_L2_TO_L1_MSGS_PER_TX),
+          newPublicDataWrites.slice(
+            i * MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX,
+            (i + 1) * MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX,
+          ),
+          newContracts.slice(i * MAX_NEW_CONTRACTS_PER_TX, (i + 1) * MAX_NEW_CONTRACTS_PER_TX),
+          newContractData.slice(i * MAX_NEW_CONTRACTS_PER_TX, (i + 1) * MAX_NEW_CONTRACTS_PER_TX),
+          newEncryptedLogs!.txLogs[i],
+          newUnencryptedLogs!.txLogs[i],
+        ),
+      );
+    }
+
+    return new this(newL1ToL2Messages, txEffects);
+  }
+
+  /**
+   * Computes the calldata hash for the L2 block
+   * This calldata hash is also computed by the rollup contract when the block is submitted,
+   * and inside the circuit, it is part of the public inputs.
+   * @returns The calldata hash.
+   */
+  getCalldataHash() {
+    const computeRoot = (leafs: Buffer[]): Buffer => {
+      const layers: Buffer[][] = [leafs];
+      let activeLayer = 0;
+
+      while (layers[activeLayer].length > 1) {
+        const layer: Buffer[] = [];
+        const layerLength = layers[activeLayer].length;
+
+        for (let i = 0; i < layerLength; i += 2) {
+          const left = layers[activeLayer][i];
+          const right = layers[activeLayer][i + 1];
+
+          layer.push(sha256(Buffer.concat([left, right])));
+        }
+
+        layers.push(layer);
+        activeLayer++;
+      }
+
+      return layers[layers.length - 1][0];
+    };
+
+    const leafs: Buffer[] = this.txEffects.map(txEffect => txEffect.hash());
+
+    return computeRoot(leafs);
+  }
+
+  public areLogsEqual(encryptedLogs: L2BlockL2Logs, unencryptedLogs: L2BlockL2Logs) {
```

Comment:
This function doesn't seem to be used.
<img width="246" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/13470840/e3b2d6df-8093-4f1b-8623-02d33171e7a2">